bsmetadata/metadata_processors.py

```diff
@@ -178,6 +178,26 @@ def process_global(self, metadata_attrs: Dict[str, Any]) -> Optional[str]:
         return "".join(["Website Description", self.cfg.metadata_key_value_sep, metadata_attrs["value"]])
 
 
+class DatasourceProcessor(MetadataProcessor):
+    """An example metadata processor for datasource types."""
+
+    def process_global(self, metadata_attrs: Dict[str, Any]) -> Optional[str]:
+        # We represent the DATASOURCE by using meaningful information of the URL.
+        # URL: http://www.example.de/2015/forum/article/21-new-project
+        # Example: example.de forum article new project
+        return "".join(["Datasource", self.cfg.metadata_key_value_sep, metadata_attrs["value"]])
+
+
+class GenerationLengthProcessor(MetadataProcessor):
+    """An example metadata processor for the text length."""
+
+    def process_global(self, metadata_attrs: Dict[str, Any]) -> Optional[str]:
+        # We represent the length of a text by the number of characters.
+        # Example: Length: 123
+
+        return "".join(["Text Length", self.cfg.metadata_key_value_sep, metadata_attrs["value"]])
```

Comment:
> And another quick thought: Did we decide to have generation length only at a global level? It might be useful to view this as a sentence-level (or `<p>` / `<div>` level, if we have HTML tags) kind of metadata.

✅ yeah, the local level was on my to-do list, and I've now added a new function to the length preprocessor to do this at the local (sentence-based) and global (text-based) level. but we can also think of a more complex splitting method than just using „dots“ to identify the relevant „parts" of a text.

Reply:
> Does the model always get the exact number? I'm not sure if we've discussed this already, but wouldn't it make sense to have some kind of bucketing (e.g., using powers of 2)?

🚧 If I remember correctly, we talked about alternative categories, e.g. small, medium, large, but not about a specific mapping between the number of characters and a category or so. So I think we decided to continue with the specific number of characters, but I would be also a fan of more abstract categories/ buckets as well.